### PR TITLE
Jetpack performance settings

### PR DESCRIPTION
**Pull request by planarvoid**
This PR adds Jetpack performance settings based on the desings. There is now a `Site accelerator` preference which opens a screen with `Faster images` and `Faster static files` and a combined `Site accelerator` switch which turns both of them on/off. 

On the main screen you see now `Site accelerator`, `Lazy load images`, `Ad-free Video Hosting` and `More`. Clicking on `More` opens a screen where there are all the mentioned options + `Jetpack search`. 

`Ad-free video hosting` is only available for Jetpack sites with a Premium or Business plan.

`Jetpack search` is available to Jetpack sites with a business plan or WP.com sites that have the flag `isJetpackSearchSupported` set to `true`. 

The easiest way to test this is comparing the options in Calypso with the same site to the options in the WPAndroid app. It happens that the Jetpack refuses to set the settings but it happens on both platforms.

To test - with a Jetpack site with a free plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button

To test - with a Jetpack site with a premium plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above
* Try setting the new options

To test - with a Jetpack site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options

To test - with a WPCom site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options

Desings:
jetpack_peformance_settings_android-1

Implementation:
jetpack_performance_settings

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java [hunk 1]
```diff
@@ -241,9 +243,26 @@
     private WPSwitchPreference mJpUseTwoFactorPref;
 
     // Speed up settings
-    private PreferenceScreen mSpeedUpYourSiteSettings;
-    private WPSwitchPreference mServeImagesFromOurServers;
     private WPSwitchPreference mLazyLoadImages;
+    private WPSwitchPreference mLazyLoadImagesNested;
+
+    // Jetpack media settings
+    private WPSwitchPreference mAdFreeHosting;
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java]: Should we rename this one to mAdFreeVideoHosting to be consistent with naming in JetpackSettingsModel?

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/WPSwitchPreference.java [hunk 2]
```diff
@@ -37,6 +44,16 @@ public WPSwitchPreference(Context context, AttributeSet attrs) {
                 mHint = array.getString(index);
             } else if (index == R.styleable.SummaryEditTextPreference_iconTint) {
                 mTint = array.getResourceId(index, R.color.neutral);
+            } else if (index == R.styleable.SummaryEditTextPreference_switchThumbTint) {
+                mThumbTint = array.getColorStateList(index);
+            } else if (index == R.styleable.SummaryEditTextPreference_switchTrackTint) {
+                mTrackTint = array.getColorStateList(index);
+            } else if (index == R.styleable.SummaryEditTextPreference_preferenceTextColor) {
+                mTextColor = array.getResourceId(index, R.color.white);
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/WPSwitchPreference.java]: We actually pulling this color from one of the libraries, so if it'll get removed at some point we might be in trouble. I suggest using `android.R.color.white` instead (in other places too).

File: WordPress/src/main/res/xml/site_settings.xml [hunk 3]
```diff
@@ -211,6 +190,137 @@
...
+            android:title="@string/site_settings_site_accelerator">
+
+            <org.wordpress.android.ui.prefs.WPSwitchPreference
+                android:id="@+id/pref_site_accelerator"
+                android:key="@string/pref_key_site_accelerator"
+                android:layout="@layout/wp_preference_layout"
+                android:title="@string/site_settings_site_accelerator"
+                app:backgroundColorChecked="@color/primary_40"
+                app:backgroundColorUnchecked="@color/neutral_40"
+                app:preferenceTextColor="@color/white"
```
Comment (khaykov) [inline WordPress/src/main/res/xml/site_settings.xml]: As I mentioned, let's switch this to `@android:color/white`

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/WPSwitchPreference.java (hunk 2 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/WPSwitchPreference.java]: ah, I didn't know that, good point 👍

**Comment by planarvoid** [comment]
Thanks for the review, @khaykov  !

> I see an option for "Add-free video hosting" on Personal plan, where it only starts from Premium.

I've fixed this. I thought that there are only three plans (free, premium and business) and I've checked that it's removed on the free plan. Now I'm hiding it for all the plans except premium and business. 

> I don't see the Site accelerator option for wpcom site on a business plan.

I'm checking Calypso and it doesn't make sense to me. We're setting Jetpack modules for a non-Jetpack site. Anyway, I'll do the same thing within the app. 

> Behavior of the "global" Site Accelerator switch is not consistent with Calypso - over there it stays enabled if at least one child option is selected, on Android it gets disabled unless both options are selected.

Good catch, I've fixed that.

> I know you followed a design, but the behavior of More button is a bit weird - sometimes there is nothing extra inside it, it just repeats options from previous screen. Wonder, maybe we should hide it where there is nothing extra to show?

It is weird, you're right. I'm now hiding the "More" menu when the "Improved search" is not showing (because it's the only extra item in the preferences. 

> Missing release notes :D

👍 😄

**Pull request merged**